### PR TITLE
implement proxy proto conditional listener

### PR DESCRIPTION
**Pull request by mediocregopher**

### CODE REVIEW

File: genapi/genapi.go [hunk 1]
```diff
@@ -423,6 +423,13 @@ func (g *GenAPI) serve(h http.Handler, addr string, doTLS bool) {
 	}
 
 	netln := net.Listener(tcpKeepAliveListener{ln.(*net.TCPListener)})
+
+	allowedProxyCIDRsStr, _ := g.ParamStr("--proxy-proto-allowed-cidrs")
```
Comment (jameshartig) [inline genapi/genapi.go:427]: Is this easier than doing `g.ParamStrs`?

File: genapi/proxy_proto.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	}, nil
+}
+
+func (p proxyListener) Accept() (net.Conn, error) {
+	c, err := p.Listener.Accept()
+	if err != nil {
+		return c, err
+	}
+
+	// The next two steps should never error, if they do something is crazy
```
Comment (jameshartig) [inline genapi/proxy_proto.go]: // The next two steps should never error, if they do, something is crazy
🐑

File: genapi/genapi.go (hunk 1 above)
Comment (mediocregopher) [inline genapi/genapi.go:427]: for setting it as an environment variable yeah. I never really figured out how to make multi-params work when set as environment vars =/

Comment (jameshartig) [inline genapi/genapi.go:427]: Makes sense, just checking.

**Pull request merged**